### PR TITLE
feat: US_46/InvisibleWallWithCamera

### PR DESCRIPTION
**Pull request by GastonGutierrezC**
## Description

We are looking for the camera to have a border that will be in line with the camera so that mario cannot move backwards and go out of the screen.

## Changes Made

- [x] Modifications of the PlayerMovementSystem.

## Checklist

- [x]  There are collisions with map objects and gravity to make it fall to the ground.
- [x]  The wall must advance together with the camera, so that it is always present.

code reviuw

@aguirrekarina 
@OropezaHugo

### CODE REVIEW

File: MarioGame/Source/Systems/PlayerMovementSystem.cs [hunk 1]
```diff
@@ -42,6 +43,9 @@ public override void Update(GameTime gameTime, IEnumerable<Entity> entities)
                 }
                 HandleUpKey(gamePadState, keyboardState, collider, animation, movement);
                 LimitSpeed(collider, collider.maxSpeed);
+
+                colliderCamera = new ColliderComponent(collider.collider.World, camera.Position.X+1f, 100, new Rectangle(100, 100, 10, 10), BodyType.Static);
```
Comment (CatCoder36) [inline MarioGame/Source/Systems/PlayerMovementSystem.cs:47]: Crear una nueva instancia de camara en cada update afecta al rendimiento, puedes intentar crearla en otro lugar

Comment (GastonGutierrezC) [reply MarioGame/Source/Systems/PlayerMovementSystem.cs:59]: Done, ahora solo se crea una vez y despues olo cambia se pocision

**Comment by aguirrekarina** [comment]
[e368eeac-443d-416e-93bf-e1faa088af36.webm](https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/101665677/e16530b8-46ab-44d5-9f13-eb1d21a8263c)

When Mario goes to the left with a jump he can leave the scene and no one looks at him

**Comment by GastonGutierrezC** [comment]
Done, now on the jumping the game check if he can not leave the scene and no one looks at him 
@aguirrekarina

**Review by aguirrekarina — APPROVED** [review]

**Review by OropezaHugo — APPROVED** [review]

**Pull request merged**